src/main/java/org/cryptomator/ui/addvaultwizard/ChooseExistingVaultController.java

```diff
@@ -94,6 +128,17 @@ public void chooseFileAndNext() {
 		}
 	}
 
+	@FXML
+	public void restoreVaultConfigWithRecoveryKey() {
+		DirectoryChooser directoryChooser = new DirectoryChooser();
```

Comment:
keep the directory chooser (ui related) here, don't hand it to the model. Only use its return value in the model.